src/main/java/edu/byu/cs/util/RepoUrlValidator.java

```diff
@@ -2,23 +2,32 @@
 
 import edu.byu.cs.autograder.GradingException;
 import org.eclipse.jgit.annotations.Nullable;
+import org.eclipse.jgit.api.CloneCommand;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import java.io.File;
+import java.util.UUID;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 public class RepoUrlValidator {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RepoUrlValidator.class);
 
-    public static boolean isValid(@Nullable String repoUrl) {
-        try {
-            clean(repoUrl);
-            return true;
-        } catch (InvalidRepoUrlException e) {
+    public static boolean isValidRepoUrl(String url) {
+        File cloningDir = new File("./tmp" + UUID.randomUUID());
+        CloneCommand cloneCommand = Git.cloneRepository().setURI(url).setDirectory(cloningDir);
```

Comment:
The whole point of #463 is to consolidate duplicated logic. This movement of code doesn't resolve that issue. This function should be gutted entirely and replaced with a call to `GitHelper:fetchRepo()` (or variant) which implements the git cloning directly.

Reply:
Agreed. This function does need to exist (to clean up the directory/handle errors/etc) but the internal logic should be delegated to GitHelper.